turtle_graphics/turtle_graphics_one_quadrant.js

```diff
@@ -0,0 +1,174 @@
+// Please refer to this one-quadrant solution as my home work. 
+// I'd appreciate it if you could take a look at the four-quadrants solution and give some advice as well. But feel free to only review and mark this one-quadrant solution. Thank you!
+const fs = require('fs');
+
+class Turtle {
+    constructor(x = 0, y = 0) {
+        this.x = x;
+        this.y = y;
+        this.initialX = x;
+        this.initialY = y;
+        this.points = [[x, y]];
+        this.directions = ['east', 'south', 'west', 'north'];
+        this.face = this.directions[0];
+    }
+
+    forward(step) {
+        let oldX = this.x;
+        let oldY = this.y;
+        if (this.face == 'east') {
+            this.x += step;
+            for (let i = 1; i <= step; i++) {
+                this.points.push([oldX + i, this.y])
+            }
+        }
+        if (this.face == 'west') {
+            this.x -= step;
+            for (let i = 1; i <= step; i++) {
```

Comment:
for repeating logic, you can look into grouping it into a function and reuse it.